pandas/io/tests/test_common.py

```diff
@@ -90,6 +90,15 @@ def test_iterator(self):
         expected.index = [0 for i in range(len(expected))]
         tm.assert_frame_equal(concat(it), expected.iloc[1:])
 
+    def test_temporary_file(self):
+        from tempfile import TemporaryFile
+        new_file = TemporaryFile("w+")
+        new_file.write("0   0")
+        new_file.flush()
+        new_file.seek(0)
+
+        dataframe = read_table(new_file, sep=r"\s+", header=None, engine="python")
+
 
```

Comment:
No need to import `read_table` for this.  `read_csv` and `read_table` are absolutely identical (except for the default delimiter).  Also, you should check the result of the `read_` operation to make sure you aren't getting garbage out.  Also make reference the issue (the one you raised).